src/nat/builder/per_user_workflow_builder.py

```
@@ -0,0 +1,798 @@
+        self._exit_stack = AsyncExitStack()
+        return self
+
+    async def __aexit__(self, *exc_details):
+
+        assert self._exit_stack is not None, "Exit stack not initialized"
+        await self._exit_stack.__aexit__(*exc_details)
+
+    def _get_exit_stack(self) -> AsyncExitStack:
+        if self._exit_stack is None:
+            raise ValueError(
+                "Exit stack not initialized. Did you forget to call `async with PerUserWorkflowBuilder() as builder`?")
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)